src/main/java/org/opensearch/knn/grpc/proto/request/search/query/KNNQueryBuilderProtoUtils.java

```diff
@@ -0,0 +1,180 @@
+            java.util.Map<String, Object> methodParameters = new java.util.HashMap<>();
+            org.opensearch.protobufs.ObjectMap objectMap = knnQueryProto.getMethodParameters();
+
+            for (java.util.Map.Entry<String, org.opensearch.protobufs.ObjectMap.Value> entry : objectMap.getFieldsMap().entrySet()) {
+                String key = entry.getKey();
+                org.opensearch.protobufs.ObjectMap.Value value = entry.getValue();
+
+                // Extract the value based on its type
+                Object javaValue = null;
+                switch (value.getValueCase()) {
+                    case INT32:
+                        javaValue = value.getInt32();
+                        break;
+                    case INT64:
+                        javaValue = value.getInt64();
+                        break;
+                    case FLOAT:
+                        javaValue = value.getFloat();
+                        break;
+                    case DOUBLE:
+                        javaValue = value.getDouble();
+                        break;
+                    case STRING:
+                        javaValue = value.getString();
+                        break;
+                    case BOOL:
+                        javaValue = value.getBool();
+                        break;
+                    // Handle other types as needed
+                    default:
+                        // Skip unsupported types
+                        continue;
+                }
+
+                methodParameters.put(key, javaValue);
+            }
+
+            builder.methodParameters(methodParameters);
+        }
+
+        // Set rescore if present
+        if (knnQueryProto.hasRescore()) {
+            // Convert the rescore KnnQueryRescore to a RescoreContext
+            KnnQueryRescore rescoreProto = knnQueryProto.getRescore();
+
+            if (rescoreProto.getKnnQueryRescoreCase() == KnnQueryRescore.KnnQueryRescoreCase.ENABLE) {
+                // If enable is true, use default rescore context, otherwise explicitly disable
+                if (rescoreProto.getEnable()) {
+                    builder.rescoreContext(RescoreContext.getDefault());
+                } else {
+                    builder.rescoreContext(RescoreContext.EXPLICITLY_DISABLED_RESCORE_CONTEXT);
+                }
+            } else if (rescoreProto.getKnnQueryRescoreCase() == KnnQueryRescore.KnnQueryRescoreCase.CONTEXT) {
+                // If context is provided, create a RescoreContext with the specified oversample factor
+                org.opensearch.protobufs.RescoreContext contextProto = rescoreProto.getContext();
+                if (contextProto.hasOversampleFactor()) {
+                    builder.rescoreContext(RescoreContext.builder().oversampleFactor(contextProto.getOversampleFactor()).build());
+                } else {
+                    builder.rescoreContext(RescoreContext.getDefault());
+                }
+            }
+        }
```

Comment:
[minor] Let's just move this to a separate private function to ensure methods are not getting too long. Please see if we can also move some other parts of this `fromProto` function in smaller private functions